### PR TITLE
Encode upload path

### PR DESCRIPTION
**Pull request by Julian1998**
## Description

Removed all unnecessary /'s in the upload path

## Related Issue

https://github.com/owncloud/phoenix/issues/1586

## Motivation and Context

Just a small bugfix

## How Has This Been Tested?

- Upload a folder using the folder upload button / drag & drop
- The upload path shouldn't contain unnecessary /'s

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: apps/files/src/FileUpload.js [hunk 1]
```diff
@@ -12,7 +12,11 @@ class FileUpload {
     const xhr = new XMLHttpRequest()
 
     // Headers
-    xhr.open(this.type, this.url + encodeURIComponent(this.path), true)
+    this.path = this.path.replace(/(^\/+)/, '')
```
Comment (DeepDiver1975) [inline apps/files/src/FileUpload.js]: Better do this in the ctor? On a second call to upload the path will be encoded again

File: apps/files/src/components/FilesAppBar.vue [hunk 2]
```diff
@@ -122,7 +122,8 @@ export default {
       return this.$route.params.item === undefined ? this.configuration.rootFolder : this.$route.params.item
     },
     url () {
-      const path = this.item === '' ? (this.configuration.rootFolder ? `${this.configuration.rootFolder}/` : '/') : `${this.item}/`
+      let path = this.item === '' ? (this.configuration.rootFolder ? `${this.configuration.rootFolder}` : '') : `${this.item}`
```
Comment (DeepDiver1975) [inline apps/files/src/components/FilesAppBar.vue]: please double check the default value of rootFolder - I think it is / - no idea if this logic is still needed

Comment (Julian1998) [reply apps/files/src/components/FilesAppBar.vue]: alright

File: apps/files/src/FileUpload.js (hunk 1 above)
Comment (Julian1998) [reply apps/files/src/FileUpload.js]: the fileupload object is getting created for every upload call anyway.
Path can have multiple values. e.g.: Images/Landscape/1.png Images/Sea/1.png - means it has to be done anyway somewhere

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUITrashbin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4607/

<p>







</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4607/

<p>




































</p></details>

File: apps/files/src/components/FilesAppBar.vue (hunk 2 above)
Comment (Julian1998) [reply apps/files/src/components/FilesAppBar.vue]: done

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4618/

<p>





































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4618/

<p>




































</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4618/

<p>























</p></details>

File: src/store/config.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ const mutations = {
     state.server = config.server
     state.auth = config.auth
     state.openIdConnect = config.openIdConnect
-    state.rootFolder = config.rootFolder === undefined ? '/' : config.rootFolder
+    state.rootFolder = config.rootFolder === undefined ? '' : config.rootFolder
```
Comment (DeepDiver1975) [inline src/store/config.js]: why this change? If I recall correctly we rely on rootfolder being / if not set ....

**Comment by LukasHirt** [comment]
Hmm, can't merge because of the requested changes. Damn 🤦‍♂
